dbcrossbar/src/clouds/gcloud/client.rs

```diff
@@ -222,6 +222,68 @@ impl Client {
         }
     }
 
+    /// Make an HTTP POST request and return the response.
+    async fn post_helper<Body>(
+        &self,
+        url: &Url,
+        body: &Body,
+    ) -> Result<reqwest::Response, ClientError>
+    where
+        Body: fmt::Debug + Serialize + std::marker::Sync,
+    {
+        trace!("POST {}", url);
+        let token = self.token().await?;
+        let wait_options = WaitOptions::default()
+            .backoff_type(BackoffType::Exponential)
+            .retry_interval(Duration::from_secs(10))
+            // Don't retry too much because we're probably classifying some
+            // permanent errors as temporary.
+            .allowed_errors(3);
+        wait(&wait_options, move || {
+            let token = token.clone();
+            async move {
+                let resp_result = self
+                    .client
+                    .post(url.as_str())
+                    .bearer_auth(token.as_str())
+                    .json(body)
+                    .send()
+                    .await;
+                match resp_result {
+                    // The HTTP request failed outright, because of something
+                    // like a DNS error or whatever.
+                    Err(err) => {
+                        // Request and timeout errors look like the kind of
+                        // things we should probably retry. But this is based on
+                        // guesswork not experience.
+                        // In general, it is not safe to retry POST requests, however for BigQuery 503 errors, we can retry.
+                        let temporary = err.is_status()
+                            && err.status() == Some(StatusCode::SERVICE_UNAVAILABLE);
```

Comment:
Can a 503 error actually occur on this branch?

Reply:
i think i misread and put the handling in the wrong place.  fixed